lib/trino-parquet/src/main/java/io/trino/parquet/ParquetTypeUtils.java

```diff
@@ -129,27 +129,23 @@ public static Optional<ColumnDescriptor> getDescriptor(List<PrimitiveColumnIO> c
 
     private static int getPathIndex(List<PrimitiveColumnIO> columns, List<String> path)
     {
-        int maxLevel = path.size();
-        int index = -1;
         for (int columnIndex = 0; columnIndex < columns.size(); columnIndex++) {
-            ColumnIO[] fields = columns.get(columnIndex).getPath();
-            if (fields.length <= maxLevel) {
```

Comment:
I don't understand the `Optimize ParquetTypeUtils#getPathIndex` commit
for example, why this check is removed.
can you perhaps move this to separate PR so i can approve this one?

Reply:
We still have a `if (fieldPath.length != path.size())` check which is supposed to accomplish the same thing.
The main difference here is that earlier `ColumnIO[] fields = columns.get(columnIndex).getPath()` would list all the objects from root to leaf for a given primitive column, while the new `String[] fieldPath = columns.get(columnIndex).getFieldPath()` will directly give us the leaf path that we are searching for. 
E.g. For a struct `test` with 2 fields `a` and `b`, when we want to read `[test, a]` path, calling `getPath` gave us three objects corresponding to `[], [test], [test, a]`. For equality check we were were looking at last element of each array (minus first element) before, now we just get the leaf path directly and do equality checks on that.

Reply:
There's no time pressure to land this one, so we can keep the commit here and wait until it's fully reviewed here.